src/DonorProfiles/resources/js/app/components/dashboard-content/index.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+const DashboardContent = () => {
+	const tabsSelector = useSelector( state => state.tabs );
+	const [ tabs, setTabs ] = useState( [] );
+	const [ dashboardContent, setDashboardContent ] = useState( [] );
+
+	useEffect( () => {
+		setTabs( Object.entries( tabsSelector ) );
+	}, [ tabsSelector ] );
+
+	useEffect( () => {
+		if ( tabs.length > 2 ) {
+			setDashboardContent( tabs.reduce( ( content, tab ) => {
```

Comment:
Can this callback be extracted to a object method? This would increase test-ability by decoupling the definition from the implementation.

Reply:
Yup! Done, ref: https://github.com/impress-org/givewp/pull/5455/commits/4bec462a51607484cfb34ee6094efc0c1df986e4